ethcore/light/src/net/mod.rs

```diff
@@ -335,6 +342,40 @@ mod id_guard {
 	}
 }
 
+/// Provides various statistics that could
+/// be used to compute costs
+struct Statistics {
+	/// Samples of peer count
+	peer_counts: VecDeque<usize>,
+}
+
+impl Statistics {
+	pub fn new() -> Self {
+		Statistics {
+			peer_counts: VecDeque::with_capacity(MOVING_SAMPLE_SIZE),
+		}
+	}
+
+	/// Add a new peer_count sample
+	pub fn add_peer_count(&mut self, peer_count: usize) {
+		while self.peer_counts.len() >= MOVING_SAMPLE_SIZE {
+			self.peer_counts.pop_front();
+		}
+		self.peer_counts.push_back(peer_count);
+	}
+
+	/// Get the average peer count from previous samples
+	pub fn avg_peer_count(&self) -> f64 {
+		let avg = self.peer_counts.iter().map(|&v| v as u64).sum::<u64>() as f64
+			/ self.peer_counts.len() as f64;
```

Comment:
Yeah it's taken care of in the line below (it's minimum 1), even though it could be refactored :p
And would it really make sense to import another package just for dividing a sum?

Reply:
I mean can we encounter division by zero?

Reply:
Ah right, sorry :/

Reply:
I suggested to use `streaming-stats` in case we want to make this `Statistics` struct general purpose and maybe extend it with `median` and other methods, but I agree it not worth adding just for an average.